project_stock/models/project_task.py

```diff
@@ -0,0 +1,236 @@
+            ("assigned", "Assigned"),
+            ("done", "Done"),
+            ("cancel", "Cancel"),
+        ],
+        compute="_compute_stock_state",
+    )
+    picking_type_id = fields.Many2one(
+        comodel_name="stock.picking.type",
+        string="Operation Type",
+        domain="[('code', '=', 'ps_operation'), ('company_id', '=', company_id)]",
+        default=_get_default_picking_type,
+        index=True,
+        check_company=True,
+    )
+    location_src_id = fields.Many2one(
+        comodel_name="stock.location",
+        string="Source Location",
+        domain="[('usage','=','internal')]",
+        default=_get_default_location_src_id,
+        index=True,
+        check_company=True,
+    )
+    location_dest_id = fields.Many2one(
+        comodel_name="stock.location",
+        string="Destination Location",
+        domain="[('usage','=','internal')]",
+        default=_get_default_location_dest_id,
+        index=True,
+        check_company=True,
+    )
+    unreserve_visible = fields.Boolean(
+        string="Allowed to Unreserve Inventory",
+        compute="_compute_unreserve_visible",
+        help="Technical field to check when we can unreserve",
+    )
+    analytic_account_id = fields.Many2one(
+        comodel_name="account.analytic.account",
+        string="Move Analytic Account",
+        help="Move created will be assigned to this analytic account",
+    )
+    analytic_line_ids = fields.One2many(
+        comodel_name="account.analytic.line",
+        inverse_name="task_id",
+        string="Analytic Lines",
+    )
+
+    def _compute_scrap_move_count(self):
+        data = self.env["stock.scrap"].read_group(
+            [("task_id", "in", self.ids)], ["task_id"], ["task_id"]
+        )
+        count_data = {item["task_id"][0]: item["task_id_count"] for item in data}
+        for item in self:
+            item.scrap_count = count_data.get(item.id, 0)
+
+    @api.depends("move_raw_ids.state")
+    def _compute_stock_state(self):
+        for task in self:
+            task.stock_state = "pending"
+            if task.move_raw_ids:
+                states = task.mapped("move_raw_ids.state")
+                for state in ("confirmed", "assigned", "done", "cancel"):
+                    if state in states:
+                        task.stock_state = state
+                        break
+
+    @api.depends("move_raw_ids", "move_raw_ids.quantity_done")
+    def _compute_unreserve_visible(self):
+        for item in self:
+            already_reserved = item.mapped("move_raw_ids.move_line_ids")
+            any_quantity_done = any([m.quantity_done > 0 for m in item.move_raw_ids])
+            item.unreserve_visible = not any_quantity_done and already_reserved
+
+    @api.model
+    def _prepare_procurement_group_vals(self, values):
+        return {"name": values["name"]}
+
+    @api.model
+    def create(self, values):
+        if not values.get("procurement_group_id"):
```

Comment:
Why do you need a procurement group?

Reply:
It's not really necessary, it's something extra that can be removed (is removed).